db.go

```diff
@@ -134,6 +163,11 @@ type TxListReply struct {
 	Err     error
 }
 
+// BlockAddrIndex represents the indexing structure for addresses.
+// It maps a hash160 to a list of transaction locations within a block that
+// either pays to or spends from the passed UTXO for the hash160.
+type BlockAddrIndex map[[ripemd160.Size]byte][]*btcwire.TxLoc
```

Comment:
What about this in btcutil?

```Go
type PubKeyHash [ripemd160.Size]byte
```

Trouble is coming up with a good name that also works for P2SH.  Does the address index contain P2SH addresses as well?